libq/xarray.c

```diff
@@ -46,7 +46,8 @@ void *xarraypush(array_t *arr, const void *ele, size_t ele_len)
 
 void xarraysort(array_t *arr, int (*compar)(const void *, const void *))
 {
-	qsort(arr->eles, arr->num, sizeof(void *), compar);
+	if (arr->eles)
```

Comment:
nit:
if (arr->eles > 1)

Reply:
`arr->eles` is a pointer to the array of elements. Do you mean `arr->eles != NULL`?

Or do you want `arr->num > 1`?

Either works. I just want to make sure we are on the same page.

Reply:
yes, sorry, I meant arr->num, because sorting 1 item, well, yeah...

Reply:
Done.